ui/utils_test.py

```diff
@@ -0,0 +1,58 @@
+"""Tests for utils methods"""
+from tempfile import NamedTemporaryFile
+
+import pytest
+
+from ui import utils
+
+
+# pylint: disable=unused-argument
+from ui.utils import write_to_file
+
+
+def x509_file_test(key_exists, cert_exists, mock_moira, settings):
+    """
+    Test that the correct error message appears if a required x509 file is missing, and that
+    mit_moira.Moira is called once if all required files exist.
+    """
+    settings.MIT_WS_PRIVATE_KEY_FILE = NamedTemporaryFile().name if key_exists else '/fake/odl/file.key'
+    settings.MIT_WS_CERTIFICATE_FILE = NamedTemporaryFile().name if cert_exists else '/fake/odl/file.cert'
+    with pytest.raises(RuntimeError) as err:
+        utils.get_moira_client()
+        if not key_exists:
+            assert settings.MIT_WS_PRIVATE_KEY_FILE in str(err)
+        if not cert_exists:
+            assert settings.MIT_WS_CERTIFICATE_FILE in str(err)
+    if not key_exists or not cert_exists:
+        assert not mock_moira.called
+    else:
+        assert mock_moira.called_once_with(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_PRIVATE_KEY_FILE)
```

Comment:
In general I think it's a good idea to avoid helper functions in tests that use a lot of boolean flags, make a lot of assertions, and try to cover a bunch of different scenarios. It tends to obscure the purpose of the tests. When I mentioned parametrization, I meant something like this:

```python
@pytest.mark.parametrize("key_file, cert_file", [
    (NamedTemporaryFile(), None),
    (None, NamedTemporaryFile()),
    (None, None),
])
def test_get_moira_client_missing_secrets(mock_moira, settings, key_file, cert_file):
    settings.MIT_WS_PRIVATE_KEY_FILE = 'bad/file/path' if not key_file else key_file.name
    settings.MIT_WS_CERTIFICATE_FILE = 'bad/file/path' if not cert_file else cert_file.name
    with pytest.raises(RuntimeError) as err:
        utils.get_moira_client()
        assert not mock_moira.called
        if key_file is None:
            assert settings.MIT_WS_PRIVATE_KEY_FILE in str(err)
        if cert_file is None:
            assert settings.MIT_WS_CERTIFICATE_FILE in str(err)


def test_get_moira_client_success(mock_moira, settings):
    """Test that a client is returned from get_moira_client"""
    tempfile1, tempfile2 = (NamedTemporaryFile(), NamedTemporaryFile())
    settings.MIT_WS_PRIVATE_KEY_FILE = tempfile1.name
    settings.MIT_WS_CERTIFICATE_FILE = tempfile2.name
    utils.get_moira_client()
    assert mock_moira.called_once_with(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_PRIVATE_KEY_FILE)
```